### ISSUE TITLE
Interface to access trace module 

### ISSUE THREAD

**Issue by rajeshkrmeena** [pulp-platform/trace_debugger#1]
I understand that the data packet is send to APB interface after aligned_8. Unable to understand interface part from host PC to APB interface because **tracer_if**  and **tracer_reg_if** is unconnected to whole design flow. 

Can you explain a little bit about interface part from host PC to APB interface. Is it through UART or other i.e. SPI ?

**Comment by bluewww** [comment]
The APB interface is just used to configure the trace module, no data is being streamed out through it.

 The tracer itself sends out his data through a [uDMA](https://github.com/pulp-platform/udma_core) plug. This solutions is specific to PULPissimo. Using an uDMA plug was used as a very low cost solution: the uDMA can stream out data in a double buffer fashion to RAM and from RAM through some peripheral such as QSPI or UART (which is too slow though). Again this redirection to RAM was done so that we can abuse the RAM as FIFO buffer with the tradeoff being that we can cause more contention on the bus. This is less of a problem on PULPissimo since there we use address interleaving and a crossbar. So we have Tracer -> uDMA -> RAM -> peripheral connected to uDMA.

**tracer_reg_if** and **tracer_if** is a part of the uDMA plug.

Anyway I'm am working on an AXI interface. Then you will be able to control how the data is being pushed out through it by configuring  the debug unit over APB. This will be a much more useful solution.

tl;dr We reuse the uDMA IP as data interface and the already existing RAM as double bufferin PULPissimo to have a very low cost trace module in PULPissimo. I'm working on a AXI interface to make this more generic. The tracer will then also become a part of [Ariane](https://github.com/pulp-platform/ariane/).
